netbox_plugin_azuread/backends.py

```diff
@@ -99,6 +99,9 @@ def _configure_access_groups(self, user, access_token):
             azure_group_names = [entry.get('displayName') for entry in azure_groups]
             AD_GROUP_MAP = PLUGIN_SETTINGS.get('AD_GROUP_MAP', {})
             for group_name in azure_group_names:
+                if group_name not in PLUGIN_SETTINGS['AD_GROUP_FILTER']:
```

Comment:
I would first check if `PLUGIN_SETTINGS['AD_GROUP_FILTER']` is defined. I can imagine `AD_GROUP_FILTER` being initialised as an empty list and then for users who upgrade but don't define a filter, suddenly all groups are filtered out because nothing is in the filter list!